diff --git a/heartbeat/iscsi b/heartbeat/iscsi
index e8caafb..cf04082 100755
--- a/heartbeat/iscsi
+++ b/heartbeat/iscsi
@@ -188,7 +188,7 @@ open_iscsi_setup() {
 #
 # open-iscsi >= "2.0-872" changed discovery semantics
 # see http://www.mail-archive.com/open-iscsi@googlegroups.com/msg04883.html
-# there's a new discovery2 command which should be used instead discovery
+# there's a new discoverydb command which should be used instead discovery
  
 open_iscsi_discovery() {
 	local output
@@ -200,8 +200,9 @@ open_iscsi_discovery() {
 
 	ocf_version_cmp "$version" "2.0-871"
 	if [ $? -eq 2 ]; then # newer than 2.0-871?
-		discovery_variant="discovery2"
-		options="-D"
+		discovery_variant="discoverydb"
+		[ "$discovery_type" = "sendtargets" ] &&
+			options="-D"
 	fi
 	cmd="$iscsiadm -m $discovery_variant -p $OCF_RESKEY_portal -t $discovery_type $options"
 	ocf_is_probe && severity=info

