--- a/heartbeat/pgsql
+++ b/heartbeat/pgsql
@@ -540,9 +540,13 @@ pgsql_validate_all() {
         return $OCF_ERR_INSTALLED;
     fi
 
-    if ! runasowner "test -w $OCF_RESKEY_pgdata"; then
-        ocf_log err "Directory $OCF_RESKEY_pgdata is not writable by $OCF_RESKEY_pgdba"
-        exit $OCF_ERR_PERM;
+    if ocf_is_probe; then
+        ocf_log info "Don't check $OCF_RESKEY_pgdata during probe"
+    else
+        if ! runasowner "test -w $OCF_RESKEY_pgdata"; then
+            ocf_log err "Directory $OCF_RESKEY_pgdata is not writable by $OCF_RESKEY_pgdba"
+            exit $OCF_ERR_PERM;
+        fi
     fi
 
     if [ -n "$OCF_RESKEY_monitor_user" -a ! -n "$OCF_RESKEY_monitor_password" ]

